addon/spaniel-engines/ember-spaniel-engine.js

```diff
@@ -31,5 +39,18 @@ export default {
         });
       });
     }
+  },
+
+  resetReadQueue() {
```

Comment:
When do we need to reset these queues individually?

I would prefer to not add additional API's speculatively. It would seem that a single method, `reset()` would suffice.